compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaClassFieldSymbol.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.compiler.api.impl.symbols;
+
+import io.ballerina.compiler.api.symbols.ClassFieldSymbol;
+import io.ballerina.compiler.api.symbols.Qualifier;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.Symbols;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BField;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.util.Flags;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static io.ballerina.compiler.api.symbols.Qualifier.FINAL;
+import static io.ballerina.compiler.api.symbols.Qualifier.PRIVATE;
+import static io.ballerina.compiler.api.symbols.Qualifier.PUBLIC;
+import static io.ballerina.compiler.api.symbols.SymbolKind.CLASS_FIELD;
+
+/**
+ * Represents a field in a class.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaClassFieldSymbol extends BallerinaObjectFieldSymbol implements ClassFieldSymbol {
+
+    public BallerinaClassFieldSymbol(CompilerContext context, BField bField) {
+        super(context, bField, CLASS_FIELD);
+    }
+
+    @Override
+    public boolean hasDefaultValue() {
+        return false;
+    }
+
+    @Override
+    public List<Qualifier> qualifiers() {
+        if (this.qualifiers != null) {
+            return this.qualifiers;
+        }
+
+        final long symFlags = this.bField.symbol.flags;
+        List<Qualifier> quals = new ArrayList<>();
+
+        if (Symbols.isFlagOn(symFlags, Flags.PUBLIC)) {
+            quals.add(PUBLIC);
+        } else if (Symbols.isFlagOn(symFlags, Flags.PRIVATE)) {
+            quals.add(PRIVATE);
+        }
+
+        if (Symbols.isFlagOn(symFlags, Flags.FINAL)) {
```

Comment:
Can't we combine this `if` check into same if-else ladder?

Reply:
That is not right, Public/Private and Final are two different things.

Reply:
In the previous one there's an if-else ladder because there won't be cases where both `public` and `private` appear on the same symbol. But there can be `public final` or `private final` combinations.

Reply:
Ops, didn't spot the combinations in the first sight. Thanks for the clarification!